modules/LocationUtils.js

```diff
@@ -74,3 +74,10 @@ export const locationsAreEqual = (a, b) =>
   a.hash === b.hash &&
   a.key === b.key &&
   valueEqual(a.state, b.state);
+
+export const shouldReplace = (location, newPath, history, newState) => {
+  const currentPath = location.pathname + location.search + location.hash;
+  const hasState = history && history.state;
+
+  return currentPath === newPath && !hasState || valueEqual(history.state, newState);
```

Comment:
`history.state` isn't part of our API. Maybe you meant `location.state`?

I believe a test would have caught this mistake. Would you mind including one with this PR? The test should make a few assertions:

1. When we use `history.link` to navigate to the same path, a new entry is not pushed onto the stack...
2. ...unless the state has changed.

Reply:
You were right. I created 4 tests to ensure that `shouldReplace` function works properly.